pandas/core/arrays/datetimelike.py

```diff
@@ -27,13 +28,55 @@
 
 from pandas.core.algorithms import checked_add_with_arr, take, unique1d
 import pandas.core.common as com
+from pandas.core.nanops import nanstd
 
 from pandas.tseries import frequencies
 from pandas.tseries.offsets import DateOffset, Tick
 
 from .base import ExtensionOpsMixin
 
 
+def _get_reduction_vals(obj, skipna):
```

Comment:
you seem to be reinventing the wheel here. we already do all of this in nanops.py for timedelta. I am not sure how this should be integrated here, but this is not the way. (meaning re-write all of the code we already have)

Reply:
Yes, this is definitely a Proof Of Concept.  For now the main question for you is if you're on board with the idea of bringing the Index reduction signatures in line with everything else

Reply:
yes about the signatures generally. but maybe let's start with min/max as more straightforward? To do what you are suggestnig here will require using nanops (the interface can certainly be here), but the implementation is already there (in nanops).